org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceExecuteCommandHandler.java

```diff
@@ -36,7 +39,7 @@
 import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
 import org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode;
 
-public class WorkspaceExecuteCommandHandler {
+public class WorkspaceExecuteCommandHandler implements IRegistryChangeListener {

```

Comment:
You can use org.eclipse.core.runtime.IRegistryEventListener. See org.eclipse.core.internal.resources.FilterTypeManager

Reply:
Listening to IRegistryEventListener as requested.